judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/TestStaticQueriesWithDefaultValues.java

```diff
@@ -0,0 +1,268 @@
+            "REQ-SYNT-003",
+            "REQ-SYNT-004",
+            "REQ-SYNT-005",
+            "REQ-TYPE-001",
+            "REQ-TYPE-002",
+            "REQ-TYPE-004",
+            "REQ-TYPE-005",
+            "REQ-TYPE-006",
+            "REQ-TYPE-007",
+            "REQ-TYPE-008",
+            "REQ-TYPE-009",
+            "REQ-MDL-001",
+            "REQ-MDL-003",
+            "REQ-ENT-001",
+            "REQ-ENT-002",
+            "REQ-ENT-004",
+            "REQ-ENT-005",
+            "REQ-ENT-009",
+            "REQ-ENT-011",
+            "REQ-EXPR-002",
+            "REQ-EXPR-004",
+            "REQ-EXPR-005",
+            "REQ-EXPR-006",
+            "REQ-EXPR-007",
+            "REQ-EXPR-008",
+            // TODO: JNG-4392 "REQ-EXPR-012",
+            "REQ-EXPR-022"
+    })
+    void testStaticQueriesWithDefaultValuesByConstantsOrExpressions() {
+        // TODO: JNG-4662: Fixing enum handling in ternary operation
+        MyEntity e1 = myEntityDao.create(MyEntity.builder().build());
+        MyEntity e2 = myEntityDao.create(MyEntity.builder()
+                .withFldBool(true)
+                .withFldDate(LocalDate.parse("1900-01-01"))
+                .withFldTime(LocalTime.parse("12:00:13"))
+                .withFldTimestamp(OffsetDateTime.parse("2020-01-01T01:11:13-12:00").atZoneSameInstant(ZoneOffset.UTC).toLocalDateTime())
+                .withFldLong(Long.valueOf(1234567890))
+                .withFldString("AAA")
+                .withFldDecimal(-1.2302)
+                .withFldEnum(MyEnum.A03)
+                .build());
+
+        MyEntity e3 = myEntityDao.create(MyEntity.builder()
+                .withFldBool(false)
+                .withFldDate(LocalDate.now())
+                .withFldTime(LocalTime.parse("15:15:01"))
+                .withFldTimestamp(OffsetDateTime.parse("2020-01-01T01:11:13+01:00").atZoneSameInstant(ZoneOffset.UTC).toLocalDateTime())
+                .withFldLong(Long.parseLong("12345678999"))
+                .withFldString("AAA")
+                .withFldDecimal(-2.2302)
+                .withFldEnum(MyEnum.A02)
+                .build());
+
+        MyEntity e4 = myEntityDao.create(MyEntity.builder()
+                .withFldBool(true)
+                .withFldDate(LocalDate.parse("2023-01-01"))
+                .withFldTime(LocalTime.parse("15:20"))
+                .withFldTimestamp(OffsetDateTime.parse("2020-01-01T01:11:13+01:00").atZoneSameInstant(ZoneOffset.UTC).toLocalDateTime())
+                .withFldLong(Long.valueOf(-16))
+                .withFldString("Lorem ipsum")
+                .withFldDecimal(999999999.9999)
+                .withFldEnum(MyEnum.A01)
+                .build());
+
+        // TODO: check s1 other derived fileds
+        Snapshot1 s1 = snapshot1Dao.create(Snapshot1.builder().build());
+        assertTrue(snapshot1Dao.queryFldMyEntity002(s1).equals(Optional.of(e3)));
+        assertTrue(snapshot1Dao.queryFldMyEntity005(s1).equals(Optional.of(e4)));
```

Comment:
Please use the following pattern:

    assertEquals(e3.getIdentifier(), snapshot1Dao.queryFldMyEntity002(s1).orElseThrow().getIdentifier());

It is generally recommended to check equality with built-in identifiers because in Java runtime the actual objects could be different.
Of course if other fields must be equal, those also should be checked.